libraries/chain/db_market.cpp

```diff
@@ -643,8 +800,11 @@ bool database::check_call_orders(const asset_object& mia, bool enable_black_swan
     bool filled_limit = false;
```

Comment:
Please add an FC_ASSERT to check_call_orders that ensures that for_new_limit_order is only true before HF 338 / 625 (see my remark on check_for_blackswan below).

Reply:
Similar as commented above, actually we need to carefully review all `assert()`s and `FC_ASSERT()`s as mentioned in https://github.com/bitshares/bitshares-core/issues/679 and https://github.com/bitshares/bitshares-core/issues/511#issuecomment-366802721, also be careful when adding new `FC_ASSERT()`s. Actually I want to get rid of all `FC_ASSERT()`s or catch them properly when processing post-transaction jobs in `_apply_block()`.

For this one, if we add a `FC_ASSERT()` here then caused a chain halt, probably would be caused by `clear_expired_feeds` or FBA buyback.

Reply:
After HF 388, check_call_orders / check_for_blackswan only operate correctly if the maker flag is false. Adding the suggested FC_ASSERT would enforce that this condition holds even if a future change mistakenly calls check_call_orders with the flag set to true.

General question (not rhetorical): what is worse, a chain halt or a chain operating with undefined and/or wrong behaviour?

Reply:
IMHO at most times a chain halt is worse, but negative impact of undefined/wrong behavior is unbounded and sometimes is even worse. So I don't have a clear answer. I slightly tend to avoid chain halt.

Reply:
I think the opposite, especially because wrong behaviour can go unnoticed for a long time.

We need more opinions. :-) @xeroc @oxarbitrage  @ryanRfox